src/collector/mod.rs

```diff
@@ -60,6 +61,70 @@ impl From<CollectorOptions> for WebSocketListenerConfig {
 	}
 }
 
+/// This type is used to distinguish different keys in the storage
+#[derive(Clone, Copy, Debug, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub(crate) enum CollectorPrefixType {
+	/// Candidate prefixed by ParachainId
+	Candidate(Option<u32>),
+	/// Relay chain block
+	Head,
+}
+
+/// Used to query storage for multiple hash types
+#[derive(Clone, Debug, Ord, PartialOrd, Eq)]
+pub(crate) struct CollectorKey {
+	/// Prefix for a specific collector key
+	pub prefix: CollectorPrefixType,
+	/// Block or a candidate hash
+	pub hash: H256,
+}
+
+impl CollectorKey {
```

Comment:
Since all members are public it is so easy to construct it, so we could not implement these at all.

Reply:
Yes, that was the initial implementation, but I've decided that having convenience create methods is good from the terms of style.